packages/cli/src/utils.ts

```diff
@@ -70,16 +70,44 @@ export async function parseConfigFilename(filename: string): Promise<string> {
   return filename;
 }
 
-export function ora(opts?: realOra.Options): realOra.Ora {
-  return realOra({ stream: process.stdout, ...opts });
+class Logger {
+  spinner: realOra.Ora;
+
+  constructor() {
+    this.spinner = realOra({ stream: process.stdout });
+  }
+
+  success(message: string) {
+    this.spinner.succeed(message);
+  }
+
+  fail(message: string, code?: number) {
```

Comment:
How about add 1 as the default value?

Reply:
There are some cases that do not need to exit the process immediately. If we specify default value, then we need to give parameter with 'undefined' value for some cases which seem not good

Reply:
Maintaining both `abort(msg, code?)` and `fail(msg)` might be more reasonable to distinguish these use-cases.

Reply:
Isn't is quite weird to have something called 'abort' in logger?

Reply:
`abort()` should be in utils instead of logger.

Reply:
still does not make sense to use spinner in abort function and move abort function outside of logger. 'fail' already means that the program should exit.

Reply:
Ok, shall we add a new parameter to tell the function not exiting immediately? The current implementation looks implicit a bit.

Reply:
done